Source/UI/3DS/ThreedsWebViewController.swift

```diff
@@ -29,24 +29,11 @@ public class ThreedsWebViewController: UIViewController {
     // MARK: - Initialization
 
     /// Initializes a web view controller adapted to handle 3dsecure.
-    public convenience init(checkoutAPIService: CheckoutAPIService, successUrl: URL, failUrl: URL) {
-        self.init(
-            checkoutAPIProtocol: checkoutAPIService,
-            successUrl: successUrl,
-            failUrl: failUrl,
-            threeDSWKNavigationHelperFactory: ThreeDSWKNavigationHelperFactory()
-        )
-    }
+    public convenience init(environment: Environment, successUrl: URL, failUrl: URL) {
+        let logger = FramesEventLogger(environment: environment, correlationID: UUID().uuidString)
```

Comment:
Injecting a new correlation ID which means the ThreedsWebViewController and Token events will have different cor ids ?

Reply:
They didn't have the same correlation ID to begin with.